engine/execution/computation/computer/computer.go

```diff
@@ -208,43 +262,23 @@ func (e *blockComputer) executeCollection(
 		colSpan.Finish()
 	}()
 
-	var (
-		events        []flow.Event
-		serviceEvents []flow.Event
-		txResults     []flow.TransactionResult
-		gasUsed       uint64
-	)
-
 	txMetrics := fvm.NewMetricsCollector()
-
 	txCtx := fvm.NewContextFromParent(blockCtx, fvm.WithMetricsCollector(txMetrics), fvm.WithTracer(e.tracer))
-
 	for _, txBody := range collection.Transactions {
-
-		txEvents, txServiceEvents, txResult, txGasUsed, err :=
-			e.executeTransaction(txBody, colSpan, txMetrics, collectionView, programs, txCtx, txIndex)
-
+		err := e.executeTransaction(txBody, colSpan, txMetrics, collectionView, programs, txCtx, txIndex, res)
 		txIndex++
-		events = append(events, txEvents...)
-		serviceEvents = append(serviceEvents, txServiceEvents...)
-		txResults = append(txResults, txResult)
-		gasUsed += txGasUsed
-
 		if err != nil {
-			return nil, nil, nil, txIndex, 0, err
+			return txIndex, err
 		}
 	}
-
+	res.AddStateSnapshot(collectionView.(*delta.View).Interactions())
```

Comment:
Not directly related to this changes, but we should stop casting to `*delta.View` and just add `Interactions()` to the interface

Reply:
I'll create a separate issue for this. we are gonna hold on to any FVM interface changes until the emulator changes are done.